Topics/Tech_Stacks/Tkinter.md

```diff
@@ -0,0 +1,123 @@
+# Tkinter
+
+## Introduction
+
+Tkinter is a Python GUI framework that comes with the Python standard library, Tkinter is cross-platform and the visual elements are rendered using native operating system elements.
+
+Tkinter is considered the go-to Python GUI framework, but GUIs built with Tkinter are often considered unaesthetically pleasing with designs often considered outdated. Given this, Tkinter is reccomended for proprietary buisness side applications.
+
+Tkinter is lightwaeight and easy to learn compared to othe GUI frameworks. This makes it a valid candidate when selecting a framework for GUI applications in Python. Especially so if the priority is to quickly build a functional application.
+
+In this introduction, we will be building a basic GUI and familiarising ourselves with some Tkinter components
+
+---
+### Prerequisites
+- Basic knowledge of Python
+- Python installed on your system
+
+
+## Core Components
+
+A Tkinter GUI has two main components
+
+1. **The base GUI window:** Windows are the main container in which all of the other GUI elements sit inside. These other elements are known as widgets, widgets can come in many forms.
+
+2. **Widgets:** All GUI elements that are inside a window, these can come in many forms, such as buttons, textboxes or labels.
+ 
+
+
+## Setting Up
+
+Lets begin by creating a new Python file and importing Tkinter
+
+```python
+import tkinter as tk
+```
+   
+And that's all there is to it, you have successfully set up Tkinter.
+
+### Creating a window
+Here is a simple example of creating a default Tkinter window, add these two lines of code to your file:
+```python
+import tkinter as tk
+window = tk.Tk()
+window.mainloop()
+```
+
+When you run the file, you should see the blank window open up. This is the blank Tkinter window
+
+The first line initialises a Tkinter window while the second line runs its "main loop" or the loop that displays the window and waits for events.
+
+### Adding widgets
+
+#### Labels
+Lets add a simple textbox to the window that says "Hello world".
+
+Between our current two lines of code, add the following two lines
+
+```python
+greeting = tk.Label(window, text="Hello world")
+greeting.pack()
+```
+
+the resulting file should now look like:
+
+```
```

Comment:
Specify the language here like was done for the code chunk above (add "python").

Reply:
Specifying code type is important